### PR TITLE
BBD-360: fix infinite scroll overlap

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: src/tags/infinite-scroll/gb-infinite-scroll.ts [hunk 1]
```diff
@@ -41,25 +42,35 @@ export class InfiniteScroll extends FluxTag<InfiniteScrollConfig>  {
     this.flux.on(Events.SORT, this.onResultsChanged);
     this.flux.on(Events.COLLECTION_CHANGED, this.onResultsChanged);
 
-    this.reset();
+    this.items = [];
+    this.loadedItems = 0;
+    this.runwayEnd = 0;
+    this.oldScroll = null;
```
Comment (effervescentia) [inline src/tags/infinite-scroll/gb-infinite-scroll.ts]: shouldn't need to initialize to null

File: src/tags/infinite-scroll/gb-infinite-scroll.ts [hunk 2]
```diff
@@ -41,25 +42,35 @@ export class InfiniteScroll extends FluxTag<InfiniteScrollConfig>  {
...
   }
 
   onResultsChanged() {
     this.reset();
-    this.onScroll();
+    this.attachRenderer();
   }
 
   onScroll() {
+    if (this.oldScroll !== this.scroller.scrollTop) this.attachRenderer();
```
Comment (effervescentia) [inline src/tags/infinite-scroll/gb-infinite-scroll.ts]: put `this.attachRenderer()` in curly braces

File: src/tags/infinite-scroll/gb-infinite-scroll.ts [hunk 3]
```diff
@@ -73,11 +84,13 @@ export class InfiniteScroll extends FluxTag<InfiniteScrollConfig>  {
         };
         (<riot.TagElement>tombstone)._tag.unmount();
         this.items.forEach((item) => item.height = item.width = 0);
-        this.onScroll();
+        this.attachRenderer();
       });
   }
 
   maybeRequestContent(renderer: Renderer) {
+    // Items NOT needed if no more items available
```
Comment (effervescentia) [inline src/tags/infinite-scroll/gb-infinite-scroll.ts]: remove comment

**Review by effervescentia — CHANGES_REQUESTED** [review]
looks good, couple of aesthetic changes

File: src/tags/infinite-scroll/gb-infinite-scroll.ts [hunk 4]
```diff
@@ -41,25 +42,36 @@ export class InfiniteScroll extends FluxTag<InfiniteScrollConfig>  {
...
 
-    this.reset();
+    this.items = [];
+    this.loadedItems = 0;
+    this.runwayEnd = 0;
     this.onResize();
   }
 
   reset() {
     this.items = [];
```
Comment (effervescentia) [inline src/tags/infinite-scroll/gb-infinite-scroll.ts:52]: can probably be inlined as `reset()` is only called in one place

**Review by effervescentia — CHANGES_REQUESTED** [review]
one last thing

File: src/tags/infinite-scroll/gb-infinite-scroll.ts (hunk 4 above)
Comment (Vic-Dev) [reply src/tags/infinite-scroll/gb-infinite-scroll.ts:52]: ok boss :)

**Review by effervescentia — APPROVED** [review]
ready to merge!

**Pull request merged**